### DISCUSSION TITLE
SDK for Rust - GreenGrass V2 IPC API Support

### DISCUSSION THREAD

**Discussion by chandar1** [awslabs/aws-sdk-rust#1126] [Q&A] [answered]
Hi All, 
I'm trying to integrate Green Grass v2 **IPC** functionality into my application using RUST
Look like the SDK for RUST does NOT yet support GG IPC

Question 1 : When's the tentative timeline to rollout Green Grass IPC APIs in the SDK for RUST ?
Question 2 : Until then, how could we use IPC functionalities of GG. What's your alternate suggestion for developers like me ?  

Thanks,

**Comment by jdisanti** [comment]
> Question 1 : When's the tentative timeline to rollout Green Grass IPC APIs in the SDK for RUST ?

I don't think there are currently any plans to do this.

> Question 2 : Until then, how could we use IPC functionalities of GG. What's your alternate suggestion for developers like me ?

Unfortunately, I'm not so familiar with GG IPC. I imagine you'll have to use the existing solutions from their documentation.

**Comment by ysaito1001** [comment, marked as answer]
Greengrass IPC support is handled by the IoT SDK team exclusively. While the team is aware of this, currently there is no official plan to support Rust (at least not yet).

**Comment by bretambrose** [comment]
Best bet is to reach out to the greengrass team on their repos.  I'll monitor this thread for +1s though.

Beyond that, would you be looking for anything in particular architecturally or design-wise?

For ex: async vs. non-async, etc...

Would an FFI-based solution be acceptable or would you strongly prefer pure-rust?
